optuna/cli.py

```diff
@@ -362,9 +362,6 @@ class _StudySetUserAttribute(_BaseCommand):
     """Set a user attribute to a study."""
 
     def add_arguments(self, parser: ArgumentParser) -> None:
-        parser.add_argument(
-            "--study", default=None, help="This argument is deprecated. Use --study-name instead."
-        )
         parser.add_argument(
             "--study-name",
             default=None,
```

Comment:
How about making `--study-name` required and removing if-statement in `take_action`?

```suggestion
            required=True,
```